nadabot/models.py

```diff
@@ -0,0 +1,234 @@
+    )
+    contract_id = models.CharField(
+        _("contract ID"),
+        max_length=100,
+        null=False,
+        help_text=_("Contract ID of the external contract that is the source of this provider.")
+    )
+    method_name = models.CharField(
+        _("method name"),
+        max_length=100,
+        null=False,
+        help_text=_("Method name of the external contract that is the source of this provider.")
+    )
+    provider_name = models.CharField(
+        _("provider name"),
+        max_length=64,
+        null=False,
+        help_text=_("Name of the provider, e.g. 'I Am Human'.")
+    )
+    description = models.TextField(
+        _("description"),
+        blank=True,
+        null=True,
+        help_text=_("Description of the provider.")
+    )
+    status = models.CharField(
+        _("status"),
+        max_length=20,
+        choices=ProviderStatus,
+        null=False,
+        help_text=_("Status of the provider.")
+    )
+    admin_notes = models.TextField(
+        _("admin notes"),
+        blank=True,
+        null=True,
+        help_text=_("Admin notes, e.g. reason for flagging or marking inactive.")
+    )
+    default_weight = models.PositiveIntegerField(
+        _("default weight"),
+        null=False,
+        help_text=_("Default weight for this provider, e.g. 100.")
+    )
+    gas = models.BigIntegerField(
+        _("gas"),
+        blank=True,
+        null=True,
+        help_text=_("Custom gas amount required.")
+    )
+    tags = models.JSONField(
+        _("tags"),
+        blank=True,
+        null=True,
+        help_text=_("Optional tags.")
+    )
+    icon_url = models.URLField(
+        _("icon URL"),
+        blank=True,
+        null=True,
+        help_text=_("Optional icon URL.")
+    )
+    external_url = models.URLField(
+        _("external URL"),
+        blank=True,
+        null=True,
+        help_text=_("Optional external URL.")
+    )
+    submitted_by = models.ForeignKey(
+        Account,
+        on_delete=models.CASCADE,
+        verbose_name=_("submitted by"),
+        max_length=100,
+        null=False,
+        help_text=_("User who submitted this provider.")
+    )
+    submitted_at_ms = models.DateTimeField(
+        _("submitted at (milliseconds)"),
+        null=False,
+        help_text=_("Timestamp of when this provider was submitted.")
+    )
+    stamp_validity_ms = models.BigIntegerField(
+        _("stamp validity"),
+        blank=True,
+        null=True,
+        help_text=_("Milliseconds that stamps from this provider are valid for before they expire.")
+    )
+    account_id_arg_name = models.CharField(
+        _("account ID argument name"),
+        max_length=100,
+        null=False,
+        help_text=_("Name of account ID argument, e.g. 'account_id' or 'accountId' or 'account'.")
+    )
+    custom_args = models.CharField(
+        _("custom args"),
+        null=True,
+        help_text=_("Custom args as Base64VecU8.")
```

Comment:
I just checked the contract again and actually when the provider is formatted in any view logic (including events), the `custom_args` are decoded into JSON. So we don't have to worry about decoding on the python side, but let's change this to a JSONField instead of CharField.